client/src/featureform/register.py

```diff
@@ -1796,7 +1796,7 @@ class ResourceClient(Registrar):
     ```
     """
 
-    def __init__(self, host=None, local=False, insecure=False, cert_path=None):
+    def __init__(self, host=None, local=False, insecure=False, cert_path=None, dry_run=False):
```

Comment:
What does dry run do? Print everything without applying? Is it just used for testing?

Reply:
Yeah, it should be replaced in the future with a "plan" argument, but this is just so we can check the inputs without it trying to run anything. It can be used outside of testing too